Bazel build doesn't work in tfio-dev docker container.

Hi folks,
I tired to use
`bazel build -c opt --copt=-match=native -s --verbose_failures //tensorflow_io/...` to build io library inside the tf-dev docker container, but this error message always pops out:
```
EOF')
SUBCOMMAND: # @libarchive//:libarchive [action 'Compiling external/libarchive/libarchive/archive_read_support_format_zip.c']
(cd /root/.cache/bazel/_bazel_root/a7ecd8237744645c5d189c197108d6d2/execroot/org_tensorflow_io && \
  exec env - \
    PATH=/miniconda/envs/tfio-dev/bin:/miniconda/condabin:/miniconda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
    TF_HEADER_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow/include \
    TF_SHARED_LIBRARY_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow \
    TF_SHARED_LIBRARY_NAME=libtensorflow_framework.so.1 \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.d '-frandom-seed=bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.o' -fPIC -iquote external/libarchive -iquote bazel-out/k8-opt/genfiles/external/libarchive -iquote bazel-out/k8-opt/bin/external/libarchive -iquote external/com_github_madler_zlib -iquote bazel-out/k8-opt/genfiles/external/com_github_madler_zlib -iquote bazel-out/k8-opt/bin/external/com_github_madler_zlib -isystem external/libarchive -isystem bazel-out/k8-opt/genfiles/external/libarchive -isystem bazel-out/k8-opt/bin/external/libarchive -isystem external/libarchive/libarchive -isystem bazel-out/k8-opt/genfiles/external/libarchive/libarchive -isystem bazel-out/k8-opt/bin/external/libarchive/libarchive -isystem external/com_github_madler_zlib -isystem bazel-out/k8-opt/genfiles/external/com_github_madler_zlib -isystem bazel-out/k8-opt/bin/external/com_github_madler_zlib '-D_GLIBCXX_USE_CXX11_ABI=0' '-match=native' '-DPLATFORM_CONFIG_H=<config.h>' -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/libarchive/libarchive/archive_read_support_format_zip.c -o bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.o)
ERROR: /root/.cache/bazel/_bazel_root/a7ecd8237744645c5d189c197108d6d2/external/libarchive/BUILD.bazel:8:1: C++ compilation of rule '@libarchive//:libarchive' failed (Exit 1): gcc failed: error executing command
  (cd /root/.cache/bazel/_bazel_root/a7ecd8237744645c5d189c197108d6d2/execroot/org_tensorflow_io && \
  exec env - \
    PATH=/miniconda/envs/tfio-dev/bin:/miniconda/condabin:/miniconda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
    TF_HEADER_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow/include \
    TF_SHARED_LIBRARY_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow \
    TF_SHARED_LIBRARY_NAME=libtensorflow_framework.so.1 \
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.d '-frandom-seed=bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.o' -fPIC -iquote external/libarchive -iquote bazel-out/k8-opt/genfiles/external/libarchive -iquote bazel-out/k8-opt/bin/external/libarchive -iquote external/com_github_madler_zlib -iquote bazel-out/k8-opt/genfiles/external/com_github_madler_zlib -iquote bazel-out/k8-opt/bin/external/com_github_madler_zlib -isystem external/libarchive -isystem bazel-out/k8-opt/genfiles/external/libarchive -isystem bazel-out/k8-opt/bin/external/libarchive -isystem external/libarchive/libarchive -isystem bazel-out/k8-opt/genfiles/external/libarchive/libarchive -isystem bazel-out/k8-opt/bin/external/libarchive/libarchive -isystem external/com_github_madler_zlib -isystem bazel-out/k8-opt/genfiles/external/com_github_madler_zlib -isystem bazel-out/k8-opt/bin/external/com_github_madler_zlib '-D_GLIBCXX_USE_CXX11_ABI=0' '-match=native' '-DPLATFORM_CONFIG_H=<config.h>' -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/libarchive/libarchive/archive_read_support_format_zip.c -o bazel-out/k8-opt/bin/external/libarchive/_objs/libarchive/archive_read_support_format_zip.pic.o)
Execution platform: @bazel_tools//platforms:host_platform
gcc: error: unrecognized command line option '-match=native'
INFO: Elapsed time: 456.119s, Critical Path: 2.82s
INFO: 2 processes: 2 local.
FAILED: Build did NOT complete successfully
```
But the previous command(`bazel build -s --verbose_failures //tensorflow_io/...`) works for me. Did I miss anything?

Thanks,
Jiacheng

Hi @CaptainDuke,
both of these commands:
```
bazel build -c opt --copt=-mtune=generic --copt=-march=x86-64 -s --verbose_failures //tensorflow_io/...
```
```
bazel build -c opt --copt=-mtune=native --copt=-mcpu=native -s --verbose_failures //tensorflow_io/...
```
don’t work in the docker container running on my Macbook,
Here are the errors:
```
ERROR: /v/tensorflow_io/cloud/BUILD:23:1: Linking of rule '//tensorflow_io/cloud:python/ops/_gcs_config_ops.so' failed (Exit 1): gcc failed: error executing command
  (cd /root/.cache/bazel/_bazel_root/a7ecd8237744645c5d189c197108d6d2/execroot/org_tensorflow_io && \
  exec env - \
    PATH=/miniconda/envs/tfio-dev/bin:/miniconda/condabin:/miniconda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
    TF_HEADER_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow/include \
    TF_SHARED_LIBRARY_DIR=/miniconda/envs/tfio-dev/lib/python3.6/site-packages/tensorflow \
    TF_SHARED_LIBRARY_NAME=libtensorflow_framework.so.1 \
  /usr/bin/gcc -shared -o bazel-out/k8-opt/bin/tensorflow_io/cloud/python/ops/_gcs_config_ops.so '-Wl,-rpath,$ORIGIN/../../../../_solib_k8/_U_S_Stensorflow_Uio_Scloud_Cpython_Sops_S_Ugcs_Uconfig_Uops.so___Utensorflow_Uio_Scloud_Skernels' '-Wl,-rpath,$ORIGIN/../../../../_solib_k8/_U@local_Uconfig_Utf_S_S_Clibtensorflow_Uframework___Uexternal_Slocal_Uconfig_Utf' -Lbazel-out/k8-opt/bin/_solib_k8/_U_S_Stensorflow_Uio_Scloud_Cpython_Sops_S_Ugcs_Uconfig_Uops.so___Utensorflow_Uio_Scloud_Skernels -Lbazel-out/k8-opt/bin/_solib_k8/_U@local_Uconfig_Utf_S_S_Clibtensorflow_Uframework___Uexternal_Slocal_Uconfig_Utf '-fuse-ld=gold' -Wl,-no-as-needed -Wl,-z,relro,-z,now -B/usr/bin -pass-exit-codes -Wl,--gc-sections -Wl,@bazel-out/k8-opt/bin/tensorflow_io/cloud/python/ops/_gcs_config_ops.so-2.params)
Execution platform: @bazel_tools//platforms:host_platform
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_Znwm' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_Znwm' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_Znwm' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_Znwm' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against 'strlen' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against 'strlen' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_ZN10tensorflow5mutex4lockEv' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against 'strlen' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against 'strlen' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against 'strlen' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_ZN10tensorflow15OpKernelContext5inputEN4absl11string_viewEPPKNS_6TensorE' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_PC32 reloc against '_ZdlPv' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: bazel-out/k8-opt/bin/tensorflow_io/cloud/kernels/libgcs_config_ops.a(gcs_config_ops.o): requires dynamic R_X86_64_32 reloc against '__gxx_personality_v0' which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
INFO: Elapsed time: 1465.057s, Critical Path: 72.13s
INFO: 2428 processes: 2428 local.
FAILED: Build did NOT complete successfully
```
Any ideas about this?

Thanks,
Jiacheng

--- a/README.md
+++ b/README.md
@@ -134,7 +134,7 @@ $ docker run -it --rm --net=host -v ${PWD}:/v -w /v tfio-dev
 $ # In Docker, configure will install TensorFlow or use existing install
 $ ./configure.sh
 $ # Build TensorFlow I/O C++. For compilation optimization flags, the default (-march=native) optimizes the generated code for your machine's CPU type. [see here](https://www.tensorflow.org/install/source#configuration_options)
-$ bazel build -c opt --copt=-match=native -s --verbose_failures //tensorflow_io/...
+$ bazel build -c opt --copt=-march=native --copt=-fPIC -s --verbose_failures //tensorflow_io/...
 $ # Run tests with PyTest, note: some tests require launching additional containers to run (see below)
 $ pytest tests/
 $ # Build the TensorFlow I/O package

